pkg/db/db.go

```diff
@@ -0,0 +1,107 @@
+package db
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/wavesplatform/gowaves/pkg/crypto"
+	"github.com/wavesplatform/gowaves/pkg/proto"
+)
+
+const (
+	heightSuffix = "h"
+)
+
+var ErrBlockOrphaned = errors.New("block orphaned")
+
+type WavesDB struct {
+	genesis crypto.Signature
+	db      *leveldb.DB
+}
+
+func (w *WavesDB) GetRaw(key []byte) ([]byte, error) {
+	return w.db.Get(key, nil)
+}
+
+func (w *WavesDB) PutRaw(key, value []byte) error {
+	return w.db.Put(key, value, nil)
+}
+
+func (w *WavesDB) DelRaw(key []byte) error {
+	return w.db.Delete(key, nil)
+}
+
+func (w *WavesDB) Has(block crypto.Signature) (bool, error) {
+	return w.db.Has(block[:], nil)
+}
+
+func (w *WavesDB) Put(block *proto.Block) error {
+	var height uint64
+
+	switch {
+	case block.BlockSignature == w.genesis:
+		height = 1
+	default:
+		parentHeight := append(block.Parent[:], []byte(heightSuffix)...)
+		has, err := w.db.Has(parentHeight, nil)
+		if err != nil {
+			return err
+		}
+		if !has {
+			return ErrBlockOrphaned
+		}
+		heightBytes, err := w.db.Get(parentHeight, nil)
+		if err != nil {
+			return err
+		}
+		height = binary.BigEndian.Uint64(heightBytes)
+		height++
+	}
+	bytes, err := block.MarshalBinary()
+	if err != nil {
+		return err
+	}
+	if err = w.db.Put(block.BlockSignature[:], bytes, nil); err != nil {
```

Comment:
No prefix for keys? How to store other keys? May be better not to call this db "waves" db?